test/test.html

```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <title>Document</title>
+</head>
+<body>
+<!-- doxray
```

Comment:
The start-of-line `^` limitation in your regex is strange to me. What's the reason for it?

Reply:
To increase my confidence since I'm not very strong in regex. I'm open to better matching patterns if you have ideas.